public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -517,15 +534,33 @@ export class LokiDatasource
   }
 
   getLogRowContext = (row: LogRowModel, options?: RowContextOptions): Promise<{ data: DataFrame[] }> => {
-    const target = this.prepareLogRowContextQueryTarget(
-      row,
-      (options && options.limit) || 10,
-      (options && options.direction) || 'BACKWARD'
-    );
+    const direction = (options && options.direction) || 'BACKWARD';
+    const { query, range } = this.prepareLogRowContextQueryTarget(row, (options && options.limit) || 10, direction);
+
+    const reverseResultIfNeeded = (result: DataQueryResponse): DataQueryResponse => {
+      if (direction === 'BACKWARD') {
+        return result;
+      }
+
+      // direction === FORWARD
+
+      return {
+        ...result,
+        data: result.data.map((frame) => {
+          const reversedFrame = new MutableDataFrame(frame);
+          reversedFrame.reverse();
+          return reversedFrame;
+        }),
+      };
+
+      return result;
+    };
+
+    // i think this can only be called from explore currently
+    const app = CoreApp.Explore;
 
-    const reverse = options && options.direction === 'FORWARD';
     return lastValueFrom(
-      this._request(RANGE_QUERY_ENDPOINT, target).pipe(
+      this._querySingle(query, range, app, 'FIXME').pipe(
```

Comment:
What about this `FIXME` refIds?

Reply:
good catch... i need a requestId there, not sure what to put there, will investigate.